app/common/renderer/lib/vendor/base.js

```diff
@@ -51,19 +72,61 @@ export class BaseVendor {
     try {
       webdriverUrl = new URL(url);
     } catch {
-      throw new Error(`${this._translate('Invalid URL:')} ${webdriverUrl}`);
+      throw new Error(`${i18n.t('Invalid URL:')} ${webdriverUrl}`);
     }
     return webdriverUrl;
   }
+
+  /**
+   * Locally save all class properties required to access the vendor server
+   *
+   * @param {Object} vendor
+   * @param {VendorProperties} vendorProperties
+   */
+  _saveProperties(vendor, {host, path, port, https, username, accessKey, headers}) {
```

Comment:
It's not the main issue (and not a very strong opinion), but we could probably remove the private prefix `_` for functions that will be called by vendor modules. I guess they might be confused if they can use `_` ones. (we already have multiple vendor implementations, so this should not be confusing  today though)